backends/qualcomm/qnn_quantizer.py

```diff
@@ -733,9 +837,199 @@ def _annotate_upsample2d(self, gm: torch.fx.GraphModule) -> None:
+        for softmax_partition in softmax_partitions:
+            softmax_node = softmax_partition.output_nodes[0]
+            input_node = softmax_node.args[0]
+
+            input_qspec_map = {}
+            input_qspec_map[input_node] = qconfig.input_activation
+
+            softmax_node.meta[QUANT_ANNOTATION_KEY] = QuantizationAnnotation(
+                input_qspec_map=input_qspec_map,
+                output_qspec=qconfig.output_activation,
+                _annotated=True,
+            )
+
+    def _annotate_pad(self, gm: torch.fx.GraphModule) -> None:
+        op_sources = [F.pad]
+        qconfig = self._get_quant_config(op_sources)
+        pad_partitions = get_source_partitions(gm.graph, op_sources)
+
+        pad_partitions = list(itertools.chain(*pad_partitions.values()))
+        for pad_partition in pad_partitions:
+            pad_node = pad_partition.output_nodes[0]
+            input_node = pad_node.args[0]
+
+            input_qspec_map = {}
+            input_qspec_map[input_node] = qconfig.input_activation
+
+            pad_node.meta[QUANT_ANNOTATION_KEY] = QuantizationAnnotation(
+                input_qspec_map=input_qspec_map,
+                output_qspec=SharedQuantizationSpec((input_node, pad_node)),
+                _annotated=True,
+            )
+
+    def _annotate_reshape(self, gm: torch.fx.GraphModule) -> None:
+        op_sources = [torch.reshape, "reshape"]
+        qconfig = self._get_quant_config(op_sources)
+        reshape_partitions = get_source_partitions(gm.graph, op_sources)
+
+        reshape_partitions = list(itertools.chain(*reshape_partitions.values()))
+        for reshape_partition in reshape_partitions:
+            reshape_node = reshape_partition.output_nodes[0]
+            input_node = reshape_node.args[0]
+
+            input_qspec_map = {}
+            input_qspec_map[input_node] = qconfig.input_activation
+
+            reshape_node.meta[QUANT_ANNOTATION_KEY] = QuantizationAnnotation(
+                input_qspec_map=input_qspec_map,
+                output_qspec=SharedQuantizationSpec((input_node, reshape_node)),
+                _annotated=True,
+            )
+
+    def _annotate_stride(self, gm: torch.fx.GraphModule) -> None:
+        op_sources = ["stride"]
+        qconfig = self._get_quant_config(op_sources)
+        stride_partitions = get_source_partitions(gm.graph, op_sources)
+
+        stride_partitions = list(itertools.chain(*stride_partitions.values()))
+        for stride_partition in stride_partitions:
+            stride_node = stride_partition.output_nodes[0]
+            input_node = stride_node.args[0]
+
+            input_qspec_map = {}
+            input_qspec_map[input_node] = qconfig.input_activation
+
+            stride_node.meta[QUANT_ANNOTATION_KEY] = QuantizationAnnotation(
+                input_qspec_map=input_qspec_map,
+                output_qspec=SharedQuantizationSpec((input_node, stride_node)),
+                _annotated=True,
+            )
```

Comment:
YOu can follow this https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py#L848 if you want to just propagate annotation for such ops.

BTW where does stride op come in?

Reply:
Thanks for your information. Actually, we will refactor out qnn quantizer in another pr. We will do this change.